pool/paymentmgr.go

```diff
@@ -538,15 +455,12 @@ func (pm *PaymentMgr) payPerShare(source *PaymentSource, amt dcrutil.Amount, hei
 		}
 	}
 	pm.setLastPaymentCreatedOn(uint64(lastPmtCreatedOn))
-	err = pm.cfg.DB.Update(func(tx *bolt.Tx) error {
-		// Update the last payment created on time and prune invalidated shares.
-		err := pm.persistLastPaymentCreatedOn(tx)
-		if err != nil {
-			return err
-		}
-		return pm.pruneShares(tx, workCreatedOn)
-	})
-	return err
+	// Update the last payment created on time and prune invalidated shares.
+	err = pm.persistLastPaymentCreatedOn()
+	if err != nil {
+		return err
+	}
+	return pruneShares(pm.cfg.DB, workCreatedOn)
```

Comment:
This ends up being two db calls instead of one, I think you should be utilizing the dbtx  there to group db calls, as was done prior.